build/buildRunTimeImages.sh

```diff
@@ -15,6 +15,140 @@ source $REPO_DIR/build/__functions.sh
 source $REPO_DIR/build/__sdkStorageConstants.sh
 source $REPO_DIR/build/__stagingRuntimeConstants.sh
 
+# Load platform versions for runtimes
+source $REPO_DIR/build/__dotNetCoreRunTimeVersions.sh
+source $REPO_DIR/build/__nodeVersions.sh
+source $REPO_DIR/build/__phpVersions.sh
+source $REPO_DIR/build/__pythonVersions.sh
+source $REPO_DIR/build/__rubyVersions.sh
+
+# Get the specific platform version used for each runtime image to assist with future patching
+# e.g., for dotnetcore:7.0-debian-buster, we would retrieve 7.0.10 from the __dotNetCoreRunTimeVersions.sh file
+function getRuntimeTagVersion()
+{
+    if [ -z $1 ]
+    then
+        echo "Runtime platform name was not provided"
+        return 1
+    fi
+
+    if [ -z $2 ]
+    then
+        echo "Runtime platform version was not provided"
+        return 1
+    fi
+
+    PLATFORM_NAME=$1
+    PLATFORM_VERSION=$2
+
+    if [ "$PLATFORM_NAME" == "dotnetcore" ]
+    then
+        case $PLATFORM_VERSION in
+
+            8.0)
+                FULL_RUNTIME_TAG_VERSION=$NET_CORE_APP_80
```

Comment:
This new method propagates the limitation that currently exists in our version selection today where we cannot release different patch versions of the same minor version to different OS types.

This means that if we stopped pushing patch versions to stretch (for example), `dotnetcore:7.0-debian-stretch` would still return the latest patch version that we have defined in the constants file.

Ideally, we can fix this in the future by writing scripts to read and sort the versionsToBuild.txt files by semver version, and then use the most recent version from in those.

Not particularly something that I think we should address here, but wanted to bring up that we likely want to rework this in the future to support different patch versions for different flavors / os types.

Reply:
Just to make sure I understand the issue: it's not _guaranteed_ that the full version we're using in each scenario is the latest (or even supported) version for a given Debian flavor, correct?

If so, then I think you're right that we'll need to read from the `versionsToBuild.txt` file to guarantee that platform versions we have available for each Debian flavor in our storage account can be referenced here, but it would require just a bit more refactoring that we may or may not need to do at the moment 😃